### PR TITLE
chore(core): change the test LUT to apply the identity function

### PR DESCRIPTION
**Pull request by IceTDrinker**
closes: https://github.com/zama-ai/tfhe-rs-internal/issues/330

### PR content/description

- the identity function more easily detects errors in the PBS as each mega case contains a different value compared to its neighbours

### Check-list:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs [hunk 1]
```diff
@@ -93,11 +93,7 @@ where
 
     let mut rsc = TestResources::new();
 
-    let f = |x: Scalar| {
-        x.wrapping_mul(Scalar::TWO)
-            .wrapping_sub(Scalar::ONE)
-            .wrapping_rem(msg_modulus)
-    };
+    let f = |x: Scalar| x.wrapping_rem(msg_modulus);
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs:96]: The approach has been to use wrapping ops in core crypto as we are on the torus, here it makes no difference, so I'm adhering to that style

and no for some parameter sets mega cases sometimes contained similar values (hiding implementation bugs), it's an issue that was reported by Baptiste from the hardware team

[the remark this answers is not in the document]
Comment (mayeul-zama) [reply tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs:96]: Indeed, mod 2 we get the same values

[the remark this answers is not in the document]
Comment (mayeul-zama) [reply tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs:96]: Wouldn't it be better not to apply the `%msg_modulus` at all ?

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs:96]: ah you are right, my bad, will remove

File: tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs [hunk 2]
```diff
@@ -93,11 +93,7 @@ where
...
 
-    let f = |x: Scalar| {
-        x.wrapping_mul(Scalar::TWO)
-            .wrapping_sub(Scalar::ONE)
-            .wrapping_rem(msg_modulus)
-    };
+    let f = |x: Scalar| x;
 
     let delta: Scalar = encoding_with_padding / msg_modulus;
     let mut msg = msg_modulus;
```
Comment (mayeul-zama) [inline tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs:99]: Could be initialized to `carry_modulus * msg_modulus`
Comment applies to other files

Comment (mayeul-zama) [reply tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs:99]: Shouldn't we not see carry and message concepts in core_crypto ?

Comment (IceTDrinker) [reply tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs:99]: strictly the carry/msg vision is a shortint vision only required when combining cleartext multiplications or ciphertexts additions with the KS + PBS, here I'm using the full plaintext space without looping operations, I don't think it's an issue as is, though I could add the carry information to parameters but keep working on the whole message space, but I don't know if we really need that at the moment, so I would probably say it's a backlog issue, wdyt ?

Comment (mayeul-zama) [reply tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs:99]: A backlog issue is fine for me

**Review by mayeul-zama — APPROVED** [review]

**Comment by IceTDrinker** [comment]
@slab-ci cpu_integer_test

File: tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs (hunk 2 above)
Comment (IceTDrinker) [reply tfhe/src/core_crypto/algorithms/test/lwe_multi_bit_programmable_bootstrapping.rs:99]: https://github.com/zama-ai/tfhe-rs-internal/issues/351

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Pull request merged**